packages/syntax-core/src/TapArea/TapArea.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { type ReactNode, forwardRef } from "react";
+import classNames from "classnames";
+import styles from "./TapArea.module.css";
+
+type TapAreaProps = {
+  /**
+   * The children to be rendered inside the tap area.
+   */
+  children?: ReactNode;
+  /**
+   * The label to be used for accessibility
+   */
+  accessibilityLabel?: string;
+  /**
+   * Test id for the tap area
+   */
+  "data-testid"?: string;
+  /**
+   * If `true`, the tap area will be disabled
+   *
+   * @defaultValue false
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the tap area will be full width
+   */
+  fullWidth?: boolean;
+  /**
+   * The callback to be called when the tap area is clicked
+   */
+  onTap: React.MouseEventHandler<HTMLDivElement>;
```

Comment:
Why did you choose to call this `onTap`? 

The touch events are all named `onTouch...` https://developer.mozilla.org/en-US/docs/Web/API/Touch_events

is `onTap` a name that will make it easier to conceptualize for the implementations in ios/android?

At first I was thinking this might be the name that `react-native` chooses to use for this kind of handler, but nope, that's [onPress](https://reactnative.dev/docs/handling-touches)

So why `onTap`?

Reply:
@benjroy because of the name **Tap**Area & because this will work across mobile and desktop.

I agree with you though so I changed it back to `onClick` - if we need `onTouch*` events, we can add those on an as-needed basis.